packages/snap/src/rpc/utxo/common/UTXOSigner.ts

```diff
@@ -31,6 +32,23 @@ export abstract class UTXOSigner<T extends UTXOChainIds> extends BaseSigner<T> {
     }
   }
 
+  async getPublicKeys({ addressParams }: GetAddressParamsType<T>): Promise<PublicKey[]> {
+    const { coin, addressNList, scriptType } = addressParams
+    try {
+      const publicKeys = await this.signer.getPublicKeys([{
+        coin,
+        addressNList,
+        curve: 'secp256k1',
+        scriptType,
+      }])
```

Comment:
That was the whole magic we needed to be able to derive pubkeys (as in, xpubs) at a specific path - no need to fiddle manually deriving from the root xpriv, the signer is an hdwallet instance, so use its `getPublicKeys` method consumed the good ol' regular way